src/wallet/wallet.cpp

```diff
@@ -289,6 +289,27 @@ void CWallet::SetBestChain(const CBlockLocator& loc)
 {
     CWalletDB walletdb(strWalletFile);
     walletdb.WriteBestBlock(loc);
+
+    UpdateConflicted();
+}
+
+void CWallet::UpdatedBlockTip(const CBlockIndex *pindex)
+{
+    UpdateConflicted();
+}
+
+void CWallet::UpdateConflicted()
+{
+    //check for conflicted transactions
+    LOCK2(cs_main, cs_wallet);
+    for (map<uint256, CWalletTx>::iterator it = mapWallet.begin(); it != mapWallet.end(); ++it)
+    {
+        CWalletTx* wtx = &((*it).second);
+
+        // update the conflicted cache flag
+        if (wtx->GetDepthInMainChain() <= 0)
```

Comment:
I don't disagree that it is okay. But it's extremely ugly (and very easily turned into an infinite loop if the logic ever gets changed).